plugins/jobs/src/js/pages/JobRunHistoryTable.js

```diff
@@ -246,17 +256,39 @@ class JobRunHistoryTable extends React.Component {
   renderJobIDColumn(prop, row, rowOptions = {}) {
     if (!rowOptions.isParent) {
       const taskID = row.taskID;
-      const id = this.props.job.id;
+
+      if (!this.state.mesosStateStoreLoaded) {
+        return <Trans>Loading...</Trans>;
+      }
+
+      // It may be that the data only *seems* to be present because we don't
+      // remove tasks from the MesosStateStore when they're garbage collected in
+      // the background.
+      const dataSeemsStillPresent = !!MesosStateStore.getTaskFromTaskID(taskID);
```

Comment:
What do you think about something like:
```
const content = !!MesosStateStore.getTaskFromTaskID(taskID) ?
  <Link
  ...
  :
  <Tooltip
  ...
```
since we never need both at the same time.

Reply:
i thought the same at some point in time but must have forgotten about it again...
it's changed!